utility_functions.h

```diff
@@ -0,0 +1,32 @@
+/////////////////////////////// INCLUDE GUARD ////////////////////////////////
+
+#ifndef included_utility_functions
+#define included_utility_functions
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+#include <ibamr/config.h>
+
+#include <CellData.h>
+#include <CellIndex.h>
+
+#include <functional>
+namespace IBAMR
+{
+/*!
+ * Computes the convolution of (alpha*a + beta*b) and the function phi at a given patch index, assuming phi has finite
+ * support.
+ *
+ * Both patch data MUST have enough ghost cells for the width of the phi function. This assumes the phi function is
+ * centered at idx and scaled by the grid spacing.
+ */
+double convolution(double alpha,
```

Comment:
Could you explain these parameters some more? I'm trying to include this in the two files I've been working on. In particular I'm curious as to where phi, phi_width, and dx come from and should be defined. I'm assuming a_data and b_data correspond loosely to phi_a and phi_b?

Reply:
I'm also a bit confused in my implementation since to compute alpha, you need to compute R2, which uses eta. But then eta is computed using alpha? This seems like a circular dependency.